igvm/hypervisor.py

```diff
@@ -555,15 +556,77 @@ def vm_block_device_name(self):
         the guest OS"""
         return 'vda1'
 
-    def migrate_vm(
-        self, vm, target_hypervisor, offline, offline_transport, transaction,
-        no_shutdown,
+    def check_migrate_vm(
+        self, vm: VM, offline: bool, offline_transport: str,
+        disk_size: int = 0,
     ):
-        if offline_transport not in ['netcat', 'drbd']:
+        if offline_transport not in ['netcat', 'drbd', 'xfs']:
             raise StorageError(
                 'Unknown offline transport method {}!'
                 .format(offline_transport)
             )
+        disk_size = disk_size if 0 < disk_size else 0
+        if not (offline and offline_transport == 'xfs') and 0 < disk_size:
+            raise StorageError(
+                'disk_size can be applied only with offline transport xfs!'
+            )
+        allocated_space = vm.dataset_obj['disk_size_gib'] - vm.disk_free()
+        if 0 < disk_size < allocated_space:
+            raise StorageError(
+                'disk_size is lower than allocated space: {} < {}!'
+                .format(disk_size, allocated_space)
+            )
+
+    def migrate_disk_size(
+        self, vm: VM, offline: bool, offline_transport: str,
+        disk_size: int = 0,
+    ) -> int:
```

Comment:
I was confused in the first place when reading the method name because I thought it would actually "migrate" or change the disk size. Maybe we can find a better name for this by inverting like e.g. `disk_size_for_migration`

Reply:
Renamed to `vm_new_disk_size`